widget/embedded/src/store/bestRoute.ts

```diff
@@ -188,63 +193,75 @@ export const useBestRouteStore = createSelectors(
       retry: (pendingSwap) => {
         const { tokens, blockchains } = useMetaStore.getState().meta;
         const connectedWallets = useWalletsStore.getState().connectedWallets;
-        const failedIndex =
-          pendingSwap.status === 'failed'
-            ? pendingSwap.steps.findIndex((s) => s.status === 'failed')
-            : null;
-
-        if (failedIndex === null || failedIndex < 0) {
-          return;
-        }
 
+        let fromBlockchain: string | null = null;
+        let fromToken: Asset | null = null;
+        let inputAmount = '';
         const firstStep = pendingSwap.steps[0];
         const lastStep = pendingSwap.steps[pendingSwap.steps.length - 1];
-        const fromBlockchain =
-          blockchains.find(
-            (blockchain) => blockchain.name === firstStep.fromBlockchain
-          ) || null;
-        const toBlockchain =
+        const lastSuccessfulStep = pendingSwap.steps
```

Comment:
We have this logic in `getLastSuccessfulStep` function in rango-preset. We could probably use that here and avoid duplicate codes.